### PR TITLE
BIGIP: NewModule.bigiq device discovery

### PR DESCRIPTION
**Pull request by wojtek0806**
##### SUMMARY
adds new module to manage BIGIP devices on the BIGIQ

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/network/f5/bigiq_device_discovery.py

### CODE REVIEW

File: lib/ansible/modules/network/f5/bigiq_device_discovery.py [hunk 1]
```diff
@@ -0,0 +1,1239 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, F5 Networks Inc.
```
Comment (caphrim007) [inline lib/ansible/modules/network/f5/bigiq_device_discovery.py]:
```suggestion
# Copyright: (c) 2019, F5 Networks Inc.
```

File: lib/ansible/modules/network/f5/bigiq_device_discovery.py [hunk 2]
```diff
@@ -0,0 +1,1239 @@
...
+version_added: 2.8
+options:
+  device_address:
+    description:
+      - The IP address of the BIG-IP device to be imported/managed.
+    type: str
+    required: True
+  device_username:
+    description:
+      - The administrator username fot the BIG-IP device.
```
Comment (caphrim007) [inline lib/ansible/modules/network/f5/bigiq_device_discovery.py]:
```suggestion
      - The administrator username for the BIG-IP device.
```

File: lib/ansible/modules/network/f5/bigiq_device_discovery.py [hunk 3]
```diff
@@ -0,0 +1,1239 @@
...
+      - This parameter is only required when adding a new BIG-IP device to be managed.
+    type: str
+  device_password:
+    description:
+      - The administrator password fot the BIG-IP device.
```
Comment (caphrim007) [inline lib/ansible/modules/network/f5/bigiq_device_discovery.py]:
```suggestion
      - The administrator password for the BIG-IP device.
```

File: lib/ansible/modules/network/f5/bigiq_device_discovery.py [hunk 4]
```diff
@@ -0,0 +1,1239 @@
...
+  returned: changed
+  type: bool
+  sample: yes
+conflict_policy:
+  description: Sets the conflict resolution policy for shared objects across BIG-IP devices.
+  returned: changed
+  type: str
+  sample: use_bigip
+device_conflict_policy:
+  description: Sets the conflict resolution policy for objects that are specific to a particular to a BIG-IP device
```
Comment (caphrim007) [inline lib/ansible/modules/network/f5/bigiq_device_discovery.py]:
```suggestion
  description: Sets the conflict resolution policy for objects that are specific to a particular to a BIG-IP device.
```

File: test/units/modules/network/f5/test_bigiq_device_discovery.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, F5 Networks Inc.
```
Comment (caphrim007) [inline test/units/modules/network/f5/test_bigiq_device_discovery.py]:
```suggestion
# Copyright: (c) 2019, F5 Networks Inc.
```

**Review by caphrim007 — CHANGES_REQUESTED** [review]
Minor fixes. Merge not contingent on their fixes. If you want to send a separate PR, lemme know and I will merge this in its current state

**Comment by wojtek0806** [comment]
@caphrim007 changes submitted ready for merge

**Review by caphrim007 — APPROVED** [review]

**Comment by caphrim007** [comment]
shipit

**Pull request merged**